x/nexus/keeper/transfer.go

```diff
@@ -104,7 +104,7 @@ func (k Keeper) EnqueueTransfer(ctx sdk.Context, senderChain exported.Chain, rec
 		return 0, fmt.Errorf("sender's chain %s does not support foreign assets", senderChain.Name)
 	}
 
-	if !recipient.Chain.SupportsForeignAssets && !(isNativeAsset && senderChain.Name == chain.Name) {
+	if !recipient.Chain.SupportsForeignAssets && !(isNativeAsset && recipient.Chain.Name == chain.Name) {
 		return 0, fmt.Errorf("recipient's chain %s does not support foreign assets", recipient.Chain.Name)
 	}
```

Comment:
This is boolean expression is quite hard to read. It took me a while understand what's going on. Can you extract it into a function like 
```
func (k Keeper) validateTransferAsset(ctx sdk.Context, chain exported.Chain, asset string) error {
	if chain.SupportsForeignAssets {
		return nil
	}

	assetChain, isNativeAsset := k.GetChainByNativeAsset(ctx, asset)
	if !isNativeAsset || assetChain.Name != chain.Name {
		return fmt.Errorf("chain %s does not support foreign assets", chain.Name)
	}

	return nil
}
```
Then the check above for the sender chain can also be replaced with the function. Later, we can also make sure that the asset is registered for the chain in the same function.